src/OrchardCore/OrchardCore.DisplayManagement.Abstractions/Utilities/StringExtensions.cs

```diff
@@ -1,14 +1,28 @@
+using System.Buffers;
+
 namespace OrchardCore.DisplayManagement.Utilities;
 
 public static class StringExtensions
 {
+    private static readonly SearchValues<char> AlternateChars = SearchValues.Create("-.");
+
     /// <summary>
     /// Encodes dashed and dots so that they don't conflict in filenames.
     /// </summary>
     /// <param name="alternateElement"></param>
     /// <returns></returns>
     public static string EncodeAlternateElement(this string alternateElement)
     {
+        if (string.IsNullOrEmpty(alternateElement))
+        {
+            return "";
+        }
+
+        if (!alternateElement.AsSpan().ContainsAny(AlternateChars))
+        {
+            return alternateElement;
+        }
```

Comment:
How does this fix an exception? Or is this condition just for perf?

Reply:
Right, it's unrelated, just trying to squeeze some perf. Also an opportunity to raise awareness about `SeachValues`.

Reply:
Yeah, that's good, TIL.